client/core/trade.go

```diff
@@ -699,15 +711,46 @@ func (t *trackedTrade) isRedeemable(match *matchTracker) bool {
 	return false
 }
 
+// shouldBeginFindRedemption will be true if we are the Taker on this match,
+// we've broadcasted a swap, our swap has gotten the required confs, we've
+// not been notified of Maker's redeem and we've not refunded our swap.
+//
+// This method accesses match fields and MUST be called with the trackedTrade
+// mutex lock held for reads.
+func (t *trackedTrade) shouldBeginFindRedemption(match *matchTracker) bool {
+	dbMatch, _, proof, _ := match.parts()
+	swapCoinID := proof.TakerSwap
+	if dbMatch.Side != order.Taker || len(swapCoinID) == 0 || len(proof.MakerRedeem) > 0 || len(proof.RefundCoin) > 0 {
+		log.Debugf(
+			"Not finding redemption for match %v: side = %s, failErr = %v, TakerSwap = %v RefundCoin = %v",
+			match.id, dbMatch.Side, match.failErr, proof.TakerSwap, proof.RefundCoin)
+		return false
+	}
+	if _, alreadyFinding := t.awaitingRedemption[swapCoinID.String()]; alreadyFinding {
+		return false
+	}
+
+	confs, err := t.wallets.fromWallet.Confirmations([]byte(swapCoinID))
+	if err != nil {
+		log.Errorf("Failed to get confirmations of the taker's swap %s (%s) for match %v, order %v",
+			coinIDString(t.wallets.fromAsset.ID, swapCoinID), t.wallets.fromAsset.Symbol, match.id, t.UID())
+		return false
+	}
+	return confs >= t.wallets.fromAsset.SwapConf
+}
+
 // isRefundable will be true if all of the following are true:
 // - We have broadcasted a swap contract (matchProof.Script != nil).
 // - Neither party has redeemed (matchStatus < order.MakerRedeemed).
 //   For Maker, this means we've not redeemed. For Taker, this means we've
-//   not been notified of Maker's redeem.
+//   not been notified of / we haven't found Maker's redeem.
```

Comment:
Where is this done though?  If the find redeem fails to find a redeem, I don't see anything to trigger a refund attempt instead.

Reply:
OK, because there will be an entry in the `awaitingRedemption` map.  This comment might make more sense as "we haven't yet found the Maker's redeem" since the search is ongoing.